### PR TITLE
Autohiss + Baydev language update.

### PR DESCRIPTION
**Pull request by tigercat2000**
First off, this PR updates most of language code to baystation code. See 11aeb63 for more details. This is necessary for autohiss to work.

Secondly, this adds a new "auto-hiss" system.
Baystation12/Baystation12#10980
>This commit adds a system (fully client-toggled) to allow unathi and
tajaran to "auto-hiss". This means that, for unathi, "S" will be
automatically extended to "Ss", "Sss", or "Ssss". Which one it changes is
based on a random pick().

>There are three modes, off, basic, full.
Off, is obviously, off. It does nothing. This is the default setting.
Basic, for unathi, enables the S extensions, and for tajaran, enables the
"R" extending.
Full, for unathi, enables the S extensions, alongside replacing "X" with
"Ks". It does not currently do anything extra for Tajaran.

>Sinta'Unathi is unaffected for Unathi, Siik'Tajr is unaffected for
Tajaran.

>On baystation, NO_STUTTER languages would also be unaffected (the noise
language). However, we do not have NO_STUTTER languages. So it hilariously
does affect the noise language. "Zartulnal Hharar dancess."

Basically, it allows players to <i>choose</i> to have the automatic purring or S extending, for convenience.

### CODE REVIEW

**Comment by Ultimate-Chimera** [comment]
Xenos should do it too.

**Comment by tigercat2000** [comment]
Not paticularly easy to implement for xenomorphs.

**Comment by Fox-McCloud** [comment]
I actually really like the Tajaran standard accent, but don't think it should be forced--as long as this is voluntary, I'm alll for it. It saves a lot of keystrokes.

:+1:

**Comment by tigercat2000** [comment]
@Fox-McCloud Yeah, the system is completely voluntary, and disabled by default.

File: code/modules/mob/living/carbon/human/say.dm [hunk 1]
```diff
@@ -257,27 +106,27 @@
 		verb = speaking.get_spoken_verb(ending)
 	else
 		if(ending == "!")
-			verb = "exclaims"
-		else if(ending == "?")
-			verb = "asks"
+			return pick("exclaims", "shouts", "yells")
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/say.dm]: I'd leave this as just "exclaims" personally, but that's about the only thing I really noticed about this PR, for now.

Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/say.dm]: If I recall correctly, the old system was broken to hell, and used the pick list anyways.

Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/say.dm]: Yeah, and there was a few we adjusted---I think this was one of them, because "!" doesn't automatically equate to yelling. *shrug*. Not a super big deal, TBH.

**Comment by Earthdivine** [comment]
>Toggleable

Oh thank god.

**Comment by Keilden** [comment]
For kidan, C should also turn into zzz if the wiki is correct.

File: code/modules/mob/living/autohiss.dm [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		return message
+	return species.handle_autohiss(message, L, client.autohiss_mode)
+
+/client
+	var/autohiss_mode = AUTOHISS_OFF
+
+/client/verb/toggle_autohiss()
+	set name = "Toggle Auto-Accent"
+	set desc = "Toggle automatic accents for your species"
+	set category = "OOC"
```
Comment (Fox-McCloud) [inline code/modules/mob/living/autohiss.dm]: Shouldn't this be an IC category? I mean, I don't care that it's OOC, just something I randomly noticed.

Comment (tigercat2000) [inline code/modules/mob/living/autohiss.dm]: *shrug* It doesn't really fit in anywhere, it's pretty much equally out of place in the IC category, the OOC category, or the Preferences category.

Comment (Fox-McCloud) [inline code/modules/mob/living/autohiss.dm]: yeah, like I said, I don't really care either; I just found it odd that it was shoved into the OOC tab when things like "character pose" and such are in the IC tab (I think).

**Comment by SamCroswell** [comment]
Miiight want to hold off on 11aeb63.
I'm told Mark has been working on a refactor of some say stuff for a little while now.

Other than that, as long as the accent is toggleable I don't see an issue with it. Perhaps we could add an accent option in character creation so that multiple accents per species are supported?

**Comment by tigercat2000** [comment]
@SamCroswell The autohiss isn't feasible to implement without 11aeb63. It relies on a few things established there.

**Comment by SamCroswell** [comment]
Then I'm sure Mark will come soon and say if it's compatible with his own changes or not.

**Comment by Markolie** [comment]
This is good to go - I haven't refactored anything.

**Comment by KasparoVy** [comment]
THANK YOU SO MUCH FOR PORTING THIS, I can't contain my excitement at seeing this PR brought over. I remember cruising the Bay Git and seeing this like 'damn that'd be wicked' but it was nothing but a fantasy. Wicked wicked stuff, super excited to see it proposed for merge. I sincerely hope it goes through

**Comment by tigercat2000** [comment]
Yay. Perrrrrrfect.
